### PR TITLE
Feature/SOF-7235: Add extension to bridge data between host and JL

### PR DESCRIPTION
**Pull request by VsevolodX**
Adds:
- extensions folder with data_bridge extension index.ts
- setup.sh for prebuilt extension development setup
- github action to build an extension

### CODE REVIEW

File: extensions/src/data_bridge/index.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        window.addEventListener("message", async (event) => {
+            if (event.data.type === "from-host-to-iframe") {
+                let data = event.data.data;
+                const dataJson = JSON.stringify(data);
+                const code = `
+  import json
+  data = json.loads('${dataJson}')
+  `;
+
+                const currentWidget = app.shell.currentWidget;
```
Comment (timurbazhirov) [inline extensions/src/data_bridge/index.ts:51]: Let's add a link to JupyterLab docs

File: extensions/src/data_bridge/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import {
+    JupyterFrontEnd,
+    JupyterFrontEndPlugin,
+} from "@jupyterlab/application";
+
+import { NotebookPanel } from "@jupyterlab/notebook";
+
+/**
+ * Initialization data for the data-bridge extension.
+ */
```
Comment (timurbazhirov) [inline extensions/src/data_bridge/index.ts:10]: Let's add a link to Jupyter Lite docs that we followed here

File: extensions/src/data_bridge/index.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            window.parent.postMessage(
+                {
+                    type: "from-iframe-to-host",
+                    requestData: true,
+                },
+                "*"
+            );
+        };
+
+        //TODO: set type for data
```
Comment (timurbazhirov) [inline extensions/src/data_bridge/index.ts:41]: `// ` pls

**Review by timurbazhirov — APPROVED** [review]

File: extensions/src/data_bridge/index.ts (hunk 1 above)
Comment (VsevolodX) [reply extensions/src/data_bridge/index.ts:53]: https://github.com/jupyterlab/jupyterlab/blob/main/packages/application/src/shell.ts

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**